rmf_fleet_adapter/src/rmf_fleet_adapter/TaskManager.cpp

```diff
@@ -920,6 +920,31 @@ void TaskManager::set_queue(
     {
       return;
     }
+
+    // Cancel pending automatic tasks that will be removed from the new
+    // assignments.
+    std::vector<std::string> new_automatic_task_ids;
```

Comment:
We can get O(1) lookup in the second for loop below if we switch this container to an `unordered_set`.

```suggestion
    std::unordered_set<std::string> new_automatic_task_ids;
```

Reply:
thanks for catching that, https://github.com/open-rmf/rmf_ros2/pull/301/commits/9a6735de09a12e9bcb40d1017652162927b69a8c